test/e2e/specs/plans/plans__signup-starter.ts

```diff
@@ -0,0 +1,186 @@
+		let userCreatedFlag = false;
+		let siteCreatedFlag = false;
+
+		beforeAll( async function () {
+			page = await browser.newPage();
+		} );
+
+		describe( 'Register as new user', function () {
+			it( 'Navigate to Signup page', async function () {
+				const loginPage = new LoginPage( page );
+				await loginPage.visit();
+				await loginPage.clickSignUp();
+			} );
+
+			it( 'Sign up as new user', async function () {
+				const userSignupPage = new UserSignupPage( page );
+				const details = await userSignupPage.signup( email, username, password );
+				userID = details.ID;
+				bearerToken = details.bearer_token;
+
+				userCreatedFlag = true;
+			} );
+		} );
+
+		describe( 'Create new site', function () {
+			let cartCheckoutPage: CartCheckoutPage;
+
+			beforeAll( async function () {
+				await BrowserManager.setStoreCookie( page );
+			} );
+
+			it( 'Select a .wordpres.com domain name', async function () {
+				const domainSearchComponent = new DomainSearchComponent( page );
+				await domainSearchComponent.search( blogName );
+				await domainSearchComponent.selectDomain( '.wordpress.com' );
+			} );
+
+			it( 'Select WordPress.com Pro plan', async function () {
+				const signupPickPlanPage = new SignupPickPlanPage( page );
+				siteDetails = await signupPickPlanPage.selectPlan( 'Starter' );
+
+				siteCreatedFlag = true;
+			} );
+
+			it( 'See secure checkout', async function () {
+				cartCheckoutPage = new CartCheckoutPage( page );
+				await cartCheckoutPage.validateCartItem( 'WordPress.com Starter' );
+			} );
+
+			it( 'Enter payment details', async function () {
+				const paymentDetails = DataHelper.getTestPaymentDetails();
+				await cartCheckoutPage.enterBillingDetails( paymentDetails );
+				await cartCheckoutPage.enterPaymentDetails( paymentDetails );
+			} );
+
+			it( 'Make purchase', async function () {
+				await cartCheckoutPage.purchase();
+			} );
+
+			it( 'Skip to dashboard', async function () {
+				const startSiteFlow = new StartSiteFlow( page );
+				await Promise.all( [
+					page.waitForNavigation( { url: /.*\/home\/.*/ } ),
+					startSiteFlow.clickButton( 'Skip to dashboard' ),
+				] );
+			} );
+		} );
+
+		describe( 'Validate WordPress.com Starter functionality', function () {
+			let sidebarComponent: SidebarComponent;
+
+			it( 'Sidebar states user is on WordPress.com Starter plan', async function () {
+				sidebarComponent = new SidebarComponent( page );
+				const plan = await sidebarComponent.getCurrentPlanName();
+				expect( plan ).toBe( 'Starter' );
+			} );
+
+			it( 'Navigate to Upgrades > Plans', async function () {
+				sidebarComponent = new SidebarComponent( page );
+				await sidebarComponent.navigate( 'Upgrades', 'Plans' );
+			} );
+
+			it( 'Plans page states user is on WordPress.com Starter plan', async function () {
+				const plansPage = new PlansPage( page, 'current' );
+				await plansPage.clickTab( 'My Plan' );
+				await plansPage.validateActivePlan( 'Starter' );
+			} );
```

Comment:
Got a test design question!

I was comparing this with the changes in #63974. In that iteration, you do deeper validation of the Plan (namely, validate storage limits.)

This made me wonder: how much should we validate plan limitations, and what is the best place to do that?

I feel like we only need to really validate how each plan affects a site in a single place -- we don't need to repeat it across upgrading vs signup, etc.

The question is: do we do it here in signup? Or do we have separate specs that validate how a plan is supposed to behave? (e.g., limited storage space, no CSS, no storage size.)

I also think we might be able to better test the plan behavior with component tests, since it seems like most places in Calypso load the plan from the redux state and then just change the component based on that. 

All that to say... I think we might be okay, at the end to end level, with just one validation that the plan has actually been set in the backend and the site state. We could do that in the sidebar or the plans activity. We probably don't need to do both? 

What are your thoughts?

Reply:
You raise a good question.

I generally agree with what you state about moving the tests down a level where possible.
I did perform additional checks in #63974 because I felt it was a low-cost way to ensure that one of the key selling point of Pro plan, the storage limit, is indeed reflected _immediately_ after the plan is upgraded. It is possible that component tests will not catch that.